src/tools/illink/Directory.Build.targets

```diff
@@ -0,0 +1,96 @@
+    <!-- Add it to the package. -->
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="@(_ReferenceAssemblies)" PackagePath="ref\$(TargetFramework)" />
+    </ItemGroup>
+  </Target>
+
+
+  <!-- Shared logic to publish multiple projects in the same
+       package. Pack doesn't support including project references
+       (https://github.com/NuGet/Home/issues/3891), so we work around
+       this by explicitly including the publish output in the
+       package. -->
+  <PropertyGroup>
+    <IncludePublishOutput Condition="'$(IncludePublishOutput)' == ''">false</IncludePublishOutput>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(IncludePublishOutput)' == 'true'">
+    <!-- Don't include the build output. Instead, we want to include the TargetFramework-specific publish output. -->
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <!-- Suppress NuGet warning for package which sets IncludeBuildOutput=false, see https://github.com/NuGet/Home/issues/8583 -->
+    <NoWarn>$(NoWarn);NU5128</NoWarn>
+    <!-- Option to include Json files in the package. Default is to just include dlls. -->
+    <IncludeJsonFilesInPackage Condition="'$(IncludeJsonFilesInPackage)' == ''">false</IncludeJsonFilesInPackage>
+    <!-- Use a NuGet extension point to publish and set up the package files. -->
+    <!-- We can't use TargetsForTfmSpecificBuildOutput, because it doesn't run when IncludeBuildOutput is false. -->
+    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);_AddPublishOutputToPackage</TargetsForTfmSpecificContentInPackage>
+  </PropertyGroup>
+
+  <Target Name="_AddPublishOutputToPackage">
+    <PropertyGroup>
+      <PublishDir>$(BaseOutputPath)$(TargetFramework)</PublishDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <ProjectsToPublish Include="$(MSBuildProjectFile)">
+        <AdditionalProperties>TargetFramework=$(TargetFramework);PublishDir=$(PublishDir)</AdditionalProperties>
+      </ProjectsToPublish>
+    </ItemGroup>
+    <!-- Clean the publish directory in case there are any left-over artifacts (publish does not work incrementally). -->
+    <ItemGroup>
+      <_FilesToDelete Remove="@(_FilesToDelete)" />
+      <_FilesToDelete Include="$(PublishDir)/*.dll" />
+      <_FilesToDelete Include="$(PublishDir)/*.json" />
+    </ItemGroup>
+    <Delete Files="@(_FilesToDelete)" />
+    <MSBuild Projects="@(ProjectsToPublish)" Targets="Publish" />
+    <ItemGroup>
+      <_PublishOutputInPackage Include="$(PublishDir)/*.dll" />
+      <_PublishOutputInPackage Include="$(PublishDir)/*.json" Condition="'$(IncludeJsonFilesInPackage)' == 'true'" />
+    </ItemGroup>
+    <!-- Sanity check. -->
+    <Error Condition="@(_PublishOutputInPackage->Count()) == 0" Text="No publish output included in package." />
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="@(_PublishOutputInPackage)" PackagePath="$(BuildOutputTargetFolder)\$(TargetFramework)" />
+    </ItemGroup>
+  </Target>
```

Comment:
We have support for that in the packaging.targets file so ideally this code would just be deleted. This might be a bit challenging, so it would be fine to do that as a follow-up.